File: lab-redes/python/multicast/servidor_multicast.py
import socket
import struct
import time

OFFSET = 22

GRUPO_MULTICAST = "230.0.0.1"
PORTA = 4446 + OFFSET

sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM, socket.IPPROTO_UDP)
sock.setsockopt(socket.IPPROTO_IP, socket.IP_MULTICAST_TTL, 2)

print(f"[Multicast] Enviando avisos para o grupo {GRUPO_MULTICAST}:{PORTA}")
for contador in range(1, 6):
    mensagem = f"Aviso #{contador}: a aula começa em {5 - contador} minuto(s)!"
    sock.sendto(mensagem.encode("utf-8"), (GRUPO_MULTICAST, PORTA))
    print(f"[Multicast] Enviado: {mensagem}")
    time.sleep(2)

sock.close()

File: lab-redes/java/multicast/ServidorMulticast.java
import java.net.*;
import java.io.IOException;

public class ServidorMulticast {
    
    static final int OFFSET = 22;

    public static void main(String[] args) throws IOException, InterruptedException {
        String grupoMulticast = "230.0.0.1";
        int porta = 4446 + OFFSET;

        InetAddress grupo = InetAddress.getByName(grupoMulticast);
        try (DatagramSocket socket = new DatagramSocket()) {
            int contador = 1;
            System.out.println("[Multicast] Enviando avisos para o grupo " + grupoMulticast + ":" + porta);
            while (contador <= 5) {
                String mensagem = "Aviso #" + contador + ": a aula começa em " + (5 - contador) + " minuto(s)!";
                byte[] dados = mensagem.getBytes();
                DatagramPacket pacote = new DatagramPacket(dados, dados.length, grupo, porta);
                socket.send(pacote);
                System.out.println("[Multicast] Enviado: " + mensagem);
                contador++;
                Thread.sleep(2000);
            }
        }
    }
}